che-incubator/che-code
# feat: Ability to apply editor configurations from a configmap

### What does this PR do?
- Adds ability ability to apply editor configurations from a configmap 
- Support for `extensions` and `settings` was added within current PR
- `settings` configurations are applied for the `Remote` scope
- `extensions` configurations are applied for the `Workspace` scope (as far as I understand, they are not supported by VS Code on the `Remote` level)

### What issues does this PR fix?


https://issues.redhat.com/browse/CRW-7258

### How to test this PR?
- Add the following configmap to your namespace:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: vscode-editor-configurations
data: 
  extensions.json: |
    {
      "recommendations": [
          "dbaeumer.vscode-eslint",
          "github.vscode-pull-request-github"
      ]
    }
  settings.json: |
    {
      "window.header": "SOME HEADER MESSAGE",
      "window.commandCenter": false,
      "workbench.colorCustomizations": {
        "titleBar.activeBackground": "#CCA700",
        "titleBar.activeForeground": "#ffffff"
      }
    }
immutable: false
```
- Start a workspace using editor image with the PR changes: `quay.io/che-incubator-pull-requests/che-code:pr-460-amd64` 
- Settings from the configmap should be placed in the `Remote` config file: `File` => `Open file` => `/checode/remote/data/Machine/settings.json`
- Extensions from the configmap should be placed in the workspace config file: `File` => `Open file` => `projects/.code-workspace`
- Extensions should be installed
- settings related to the editor header should be applied

<img width="1054" alt="Screenshot 2024-12-16 at 15 55 41" src="https://github.com/user-attachments/assets/a22983e3-2321-4f4f-8866-991b44766013" />

### Does this PR contain changes that override default upstream Code-OSS behavior?
